src/Domain/CoordinateMaps/TimeDependent/Shape.cpp

```diff
@@ -67,15 +71,34 @@ template <typename T>
 std::array<tt::remove_cvref_wrap_t<T>, 3> Shape::operator()(
     const std::array<T, 3>& source_coords, const double time,
     const FunctionsOfTimeMap& functions_of_time) const {
-  ASSERT(functions_of_time.find(f_of_t_name_) != functions_of_time.end(),
+  ASSERT(functions_of_time.find(shape_f_of_t_name_) != functions_of_time.end(),
          "Could not find function of time: '"
-             << f_of_t_name_ << "' in functions of time. Known functions are "
+             << shape_f_of_t_name_
+             << "' in functions of time. Known functions are "
              << keys_of(functions_of_time));
 
   const auto centered_coords = center_coordinates(source_coords);
   auto theta_phis = cartesian_to_spherical(centered_coords);
   const auto interpolation_info = ylm_.set_up_interpolation_info(theta_phis);
-  const DataVector coefs = functions_of_time.at(f_of_t_name_)->func(time)[0];
+  DataVector coefs = functions_of_time.at(shape_f_of_t_name_)->func(time)[0];
+  if (size_f_of_t_name_.has_value()) {
+    ASSERT(functions_of_time.find(size_f_of_t_name_.value()) !=
+               functions_of_time.end(),
+           "Could not find function of time: '"
+               << size_f_of_t_name_.value()
+               << "' in functions of time. Known functions are "
+               << keys_of(functions_of_time));
+    ASSERT(coefs[0] == 0.0,
+           "When using a size function of time, the l=0 component of the shape "
+           "function of time must be zero. Currently it is "
+               << coefs[0]);
+    const double l0m0_spherical_harmonic_coef =
+        functions_of_time.at(size_f_of_t_name_.value())->func(time)[0][0];
+    // Size holds the *actual* \lambda_00 spherical harmonic coefficient, but
+    // shape holds Spherepack coefficients so we must convert between the two.
+    // Need to multiply lambda_00 by sqrt(2/pi)
+    coefs[0] = M_SQRT1_2 * M_2_SQRTPI * l0m0_spherical_harmonic_coef;
```

Comment:
Maybe factor into a separate function to avoid repeating this code. I think all these functions basically have two parts: 1. Get coeffs, 2. Compute expansion. The first part could be factored out.